src/sign_msg.c

```diff
@@ -11,8 +20,84 @@ typedef struct {
 
 static msg_context_t msg_context;
 
-static uint8_t set_result_signature();
-bool sign_message(void);
+static bool sign_message(void);
+
+void init_msg_context(void) {
+    bip32_account = 0;
+    bip32_address_index = 0;
+
+    app_state = APP_STATE_IDLE;
+}
+
+static uint8_t set_result_signature() {
+    uint8_t tx = 0;
+    G_io_apdu_buffer[tx++] = MESSAGE_SIGNATURE_LEN;
+    memmove(G_io_apdu_buffer + tx, msg_context.signature, MESSAGE_SIGNATURE_LEN);
+    tx += MESSAGE_SIGNATURE_LEN;
+    return tx;
+}
+
+#if defined(TARGET_STAX)
+
+static void start_review(void);
+static void ui_sign_message_nbgl(void);
+static void rejectUseCaseChoice(void);
+static nbgl_layoutTagValueList_t layout;
+static nbgl_layoutTagValue_t infos[1];
+
+static const nbgl_pageInfoLongPress_t review_final_long_press = {
+    .text = "Sign message on\nMultiversX network?",
+    .icon = &C_icon_multiversx_logo_64x64,
+    .longPressText = "Hold to sign",
+    .longPressToken = 0,
+    .tuneId = TUNE_TAP_CASUAL,
+};
+
+static void review_final_callback(bool confirmed) {
+    if (confirmed) {
+        int tx = set_result_signature();
+        io_seproxyhal_io_heartbeat();
+        send_response(tx, true, false);
+        nbgl_useCaseStatus("MESSAGE\nSIGNED", true, ui_idle);
+    } else {
+        rejectUseCaseChoice();
+    }
+}
+
+static void rejectChoice(void) {
+    send_response(0, false, false);
+    nbgl_useCaseStatus("Message\nrejected", false, ui_idle);
+}
+
+static void rejectUseCaseChoice(void) {
+    nbgl_useCaseConfirm("Reject message?", NULL, "Yes, reject", "Go back to message", rejectChoice);
```

Comment:
Maybe you can keep with snake_case here for function names?
Also reordering them avoid having to pre-define `rejectUseCaseChoice`

Reply:
Renamed + reordered + moved to utils